planner/core/find_best_task.go

```diff
@@ -343,15 +343,18 @@ func (ds *DataSource) skylinePruning(prop *property.PhysicalProperty) []*candida
 		var currentCandidate *candidatePath
 		if path.isTablePath {
 			currentCandidate = ds.getTableCandidate(path, prop)
-		} else if len(path.accessConds) > 0 || !prop.IsEmpty() || path.forced || isCoveringIndex(ds.schema.Columns, path.fullIdxCols, path.fullIdxColLens, ds.tableInfo.PKIsHandle) {
-			// We will use index to generate physical plan if any of the following conditions is satisfied:
-			// 1. This path's access cond is not nil.
-			// 2. We have a non-empty prop to match.
-			// 3. This index is forced to choose.
-			// 4. The needed columns are all covered by index columns(and handleCol).
-			currentCandidate = ds.getIndexCandidate(path, prop)
 		} else {
-			continue
+			coveredByIdx := isCoveringIndex(ds.schema.Columns, path.fullIdxCols, path.fullIdxColLens, ds.tableInfo.PKIsHandle)
```

Comment:
We cannot take the benefit of short circuit evaluation after this change. We would call `isCoveringIndex` for all of the index paths now.

Reply:
Actually there is no short circuit in the index paths, because the isCoveringIndex will be called in `getIndexCandidate` anyway.

Reply:
Make sense.